### ISSUE TITLE
investigate dropping Java 8 dependency and only use Java 7 (for nio)

### ISSUE THREAD

**Issue by cvogt** [cvogt/cbt#29]
currently NailgunLauncher uses lambda

**Comment by cvogt** [comment]
I believe #52 fixes this and drops the JDK7 requirement @tobias-johansson @farmdawgnation can you confirm?

**Comment by tobias-johansson** [comment]
Guessing you mean the JDK**8** requirement, right?
I'm on 5488e70 and can't bootstrap on JDK7 (after disabling the version check in the launcher script)

BootstrapScala.java is still java 8

```bash
> javac -version 
javac 1.7.0_95
> cbt run
(Note: nailgun not found. It makes CBT faster! Try 'brew install nailgun'.)
Recompiling CBT BootstrapScala.java
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:47: error: illegal start of expression
        Arrays.stream(ds).map(d -> d.path.toString()).toArray(String[]::new)
                                 ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:47: error: '.class' expected
        Arrays.stream(ds).map(d -> d.path.toString()).toArray(String[]::new)
                                                                      ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:47: error: ')' expected
        Arrays.stream(ds).map(d -> d.path.toString()).toArray(String[]::new)
                                                                       ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:47: error: ')' expected
        Arrays.stream(ds).map(d -> d.path.toString()).toArray(String[]::new)
                                                                        ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:47: error: ';' expected
        Arrays.stream(ds).map(d -> d.path.toString()).toArray(String[]::new)
                                                                           ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:48: error: illegal start of expression
      )
      ^
/home/tobias/Projects/cbt/bootstrap_scala/BootstrapScala.java:48: error: ';' expected
      )
       ^
7 errors
Error: Could not find or load main class BootstrapScala
Recompiling NailgunLauncher. Detected source changes.
Stopping nailgun
Restarting nailgun
Recompiling Stage1. Detected source changes.
Error: Could not find or load main class scala.tools.nsc.Main
```

**Comment by cvogt** [comment]
ok, that makes sense. Forgot about BootstrapScala.java. Ok, then we'll leave this open for now.

**Comment by tobias-johansson** [comment]
Working on fixing BootstrapScala.java

**Comment by cvogt** [comment]
fixed by #56
